pkg/redirectpolicy/manager_test.go

```diff
@@ -182,9 +182,11 @@ var (
 			Conditions: []slimcorev1.PodCondition{podReady},
 		},
 	}
-	pod1ID = k8s.ServiceID{
-		Name:      pod1.Name,
-		Namespace: pod1.Namespace,
+	pod1ID = podID{
+		ServiceID: k8s.ServiceID{
+			Name:      pod1.Name,
+			Namespace: pod1.Namespace,
+		},
```

Comment:
While I'm not opposed to adding tests, I'm afraid that in order to fully test out the scenarios within https://github.com/cilium/cilium/issues/30409, it would not suffice in a single package unit test nor would it be easy to write an e2e test for it, in order to simulate the correct conditions. The alternative is to write a controlplane test, but I'm also afraid that that framework is not quite up to spec to support the testing needed, i.e. creating endpoints, simulating delays in k8s events, etc. and would be non-trivial to do so. I'm hesitant to block the PR on such things. I also understand that it's not ideal to have a PR without tests.

On the plus side, the changes are relatively easy to understand especially regarding the LRP code, which essentially is updated to query an internal map with a more unique key, instead.

Reply:
My worry is that if we let the PR through without tests they will never be added. I feel that a separate TODO item for adding tests would go stale or become de-prioritized and never get done. I agree that the changes are relatively easy to understand, and delaying the PR isn't great, but I think it's a better alternative to merging these changes without the associated regression tests. This is a great opportunity to expand on the controlplane test framework to support testing these kinds of changes.

Reply:
I've greatly simplified the PR as per https://github.com/cilium/cilium/pull/31605#issuecomment-2097483504, so this code highlighted here no longer exists.

Reply:
After syncing offline, given the multi month effort investment required to allow the controlplane framework to have the ability to test this kind of behavior, I am resolving this thread to proceed with the PR as it resolves problematic behavior for users today.